include/oneapi/dpl/experimental/kt/internal/esimd_radix_sort_dispatchers.h

```diff
@@ -227,9 +234,80 @@ __onesweep(sycl::queue __q, _RngPack&& __pack, ::std::size_t __n)
     constexpr ::std::uint32_t __global_hist_item_count = __bin_count * __stage_count;
     const ::std::uint32_t __group_hist_item_count = __bin_count * __stage_count * __sweep_work_group_count;
 
+    // TODO: check if it is more performant to fill it inside the histgogram kernel
+    // This line assumes that global and group histograms are stored sequentially
+    sycl::event __event_chain = __q.memset(__mem_holder.__global_hist_ptr(), 0,
+                                           (__global_hist_item_count + __group_hist_item_count) * sizeof(_GlobalHistT));
+
+    // TODO: consider adding a more versatile API, e.g. passing special kernel_config parameters for histogram computation
+    constexpr ::std::uint32_t __hist_work_group_count = 64;
+    constexpr ::std::uint32_t __hist_work_group_size = 64;
+    __event_chain = __radix_sort_histogram_submitter<__is_ascending, __radix_bits, __hist_work_group_count,
+                                                     __hist_work_group_size, _EsimdRadixSortHistogram>()(
+        __q, __input_pack.__keys_rng(), __mem_holder.__global_hist_ptr(), __n, __event_chain);
+
+    __event_chain = __radix_sort_onesweep_scan_submitter<__stage_count, __bin_count, _EsimdRadixSortScan>()(
+        __q, __mem_holder.__global_hist_ptr(), __n, __event_chain);
+
+    __event_chain = __radix_sort_onesweep_submitter<__is_ascending, __radix_bits, __data_per_work_item,
+                                                    __work_group_size, _EsimdRadixSortSweep>()(
+        __q, __input_pack, __virt_pack1, __mem_holder.__global_hist_ptr(), __mem_holder.__group_hist_ptr(),
+        __sweep_work_group_count, __n, 0, __event_chain);
+
+    for (::std::uint32_t __stage = 1; __stage < __stage_count; __stage++)
+    {
+        _GlobalHistT* __p_global_hist = __mem_holder.__global_hist_ptr() + __bin_count * __stage;
+        _GlobalHistT* __p_group_hists =
+            __mem_holder.__group_hist_ptr() + __sweep_work_group_count * __bin_count * __stage;
+
+        if (__stage % 2 != 0)
+        {
+            __event_chain = __radix_sort_onesweep_submitter<__is_ascending, __radix_bits, __data_per_work_item,
+                                                            __work_group_size, _EsimdRadixSortSweep>()(
+                __q, __virt_pack1, __virt_pack2, __p_global_hist, __p_group_hists, __sweep_work_group_count, __n,
+                __stage, __event_chain);
+        }
+        else
+        {
+            __event_chain = __radix_sort_onesweep_submitter<__is_ascending, __radix_bits, __data_per_work_item,
+                                                            __work_group_size, _EsimdRadixSortSweep>()(
+                __q, __virt_pack2, __virt_pack1, __p_global_hist, __p_group_hists, __sweep_work_group_count, __n,
+                __stage, __event_chain);
+        }
```

Comment:
Unless I miss something obvious, these two kernel submissions will not generate identical kernels, and so should really have distinct names (like with the Even/Odd suffixes before the patch).

Reply:
Now in hindsight, I think you are correct, because the types of `__virt_pack1` and `__virt_pack2` may not match.

However, if they do match, do you agree that these would generate an identical kernel?

My proposal would be to merely include the types of the input & output in the kernel name, that way they aren't required to generate two identical kernels with different names when the types match.

Reply:
I'm doing some testing with no unnamed lambda, and seeing the issue.  I'll provide an update when it is fixed.

Reply:
I've fixed the issue as proposed by including the decayed RngPack typenames in the kernel name.  
I've also fixed kernel name clashing in the tests for this and the in-place version of esimd radix sort, and added a test utility to generate a new kernel_params struct with a new  kernelname.

Thanks for catching this.